src/App.tsx

```diff
@@ -15,58 +15,83 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  handleClearSelection = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  handleAdd = (good: string) => {
+    this.setState({ selectedGood: good });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+  render(): React.ReactNode {
+    const { selectedGood } = this.state;
+
+    return (
+      <main className="section container">
+        {selectedGood?.length ? (
```

Comment:
Since selectedGood is a string - there is no need to check for .length, will be quite enough to validate it like `!!selectedGood` (or, even as simple as just `selectedGood`).